src/gurt/telemetry.c

```diff
@@ -1075,6 +1075,49 @@ d_tm_compute_histogram(struct d_tm_node_t *node, uint64_t value)
 	return rc;
 }
 
+/**
+ * Set the given counter to the specified \a value
+ *
+ * \param[in,out]	metric	Pointer to the metric
+ * \param[in]		value	Increments the counter by this \a value
+ */
+void
+d_tm_set_counter(struct d_tm_node_t *metric, uint64_t value)
+{
+	if (unlikely(d_tm_shmem_root == NULL || metric == NULL))
+		return;
+
+	if (unlikely(metric->dtn_type != D_TM_COUNTER)) {
+		D_ERROR("Failed to set counter [%s] on item not a "
+			"counter.\n", metric->dtn_name);
+		return;
+	}
+
+	metric->dtn_metric->dtm_data.value = value;
+}
+
+/**
+ * Increment the given counter by the specified \a value
+ *
+ * \param[in,out]	metric	Pointer to the metric
+ * \param[in]		value	Increments the counter by this \a value
+ */
+void
+d_tm_inc_counter(struct d_tm_node_t *metric, uint64_t value)
```

Comment:
Any reason not to reuse (or refactor) d_tm_increment_counter below? Seems like this is guaranteed to cause confusion later.

Reply:
My intent was to leave the cleanup of all the test cases to you :)

Reply:
Fair enough. Is d_tm_increment_counter not being used outside the tests, then?

Reply:
I will push a follow-on patch on this one is landed to move all "real" use of  d_tm_increment_counter() to  d_tm_inc_counter(), so that you can fix the tests and remove  d_tm_increment_counter() completely.